src/convertToObject.js

```diff
@@ -6,7 +6,34 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesList = sourceString.replaceAll('\n', ' ').split(';');
+  const filtredStylesList = stylesList.filter((val) => val.trim().length > 0);
+
+  const segregatedList = [];
+  const stylesObject = {};
+
+  function segregate(el) {
+    const splited = el.split(':');
+
+    for (let i = 0; i < splited.length; i++) {
+      if (splited[i].length > 0) {
+        segregatedList.push(splited[i].trim());
+      }
+    }
+  }
+
+  function makeSegregatedObject(el, index, arr) {
+    for (let i = 0; i < arr.length; i++) {
+      if (i % 2 === 0) {
+        stylesObject[arr[i]] = arr[i + 1].replaceAll(', ', ',\n');
```

Comment:
The code assumes that every style has a value. To handle cases where a style might not have a value (e.g., 'color:'), add a check to ensure that `arr[i + 1]` is not undefined before assigning it to `stylesObject[arr[i]]`.